src/lib/icon/icon.spec.ts

```diff
@@ -207,6 +207,25 @@ describe('MatIcon', () => {
       verifyPathChildElement(svgChild, 'moo');
     });
 
+    it('should never parse the same icon set multiple times', () => {
+      // Normally we avoid spying on private methods like this, but the parsing is a private
+      // implementation detail that should not be exposed to the public API. This test, though,
+      // is important enough to warrant the brittle-ness that results.
+      spyOn(iconRegistry, '_svgElementFromString' as any).and.callThrough();
```

Comment:
Rather than spying on a private method, you should be able to assert the same by reproviding the `DOCUMENT` token and spying on `createElement`.

Reply:
I think long-term the "right" thing to do is to break SVG parsing into its own utility and then use a mock for `MatIconRegistry`, which would make sense to do as part of #5188